bestbook/api/books.py

```diff
@@ -61,108 +61,107 @@ class BookGraph(core.Base):
     # What about learning objectives?
     # What about types of edges?
 
-    __tablename__ = "bookgraph"
+    __tablename__ = "tournament_graph"
     __table_args__ = (
         PrimaryKeyConstraint(
-            'submitter', 'winner_work_olid', 'competitor_work_olid', 'topic_id'
+            'submitter', 'winner_work_olid', 'contender_work_olid', 'topic_id'
         ),
         UniqueConstraint(
-            'submitter', 'winner_work_olid', 'competitor_work_olid', 'topic_id',
+            'submitter', 'winner_work_olid', 'contender_work_olid', 'topic_id',
             name='_book_edge_uc'
         )
     )
 
     submitter = Column(Unicode, nullable=False) # e.g. @cdrini - Open Library username
     winner_work_olid = Column(Unicode, nullable=False) # Open Library ID (required)
-    # rename to contender_work_olid
-    competitor_work_olid = Column(Unicode, nullable=False) # Open Library ID (required)
+    contender_work_olid = Column(Unicode, nullable=False) # Open Library ID (required)
     topic_id = Column(Integer, ForeignKey("topics.id")) # TBBO what?
-    # review_id = Column(BigInteger, ForeignKey("reviews.id"))
-    annotation = Column(Unicode) # Free-form answer  # XXX rm
+    review_id = Column(BigInteger, ForeignKey("reviews.id"))
     created = Column(DateTime(timezone=False), default=datetime.utcnow, nullable=False)
 
     topic = relationship("Topic")
+    review = relationship("Review")
 
+    @classmethod
+    def get_distinct_topics(cls):
+        topic_ids = [t.topic_id for t in cls.query.distinct(cls.topic_id)]
+        return Topic.query.filter(Topic.id.in_(topic_ids)).all()
 
-class Book(core.Base):
 
-    __tablename__ = "books"
-    __table_args__ = (UniqueConstraint('work_olid', 'edition_olid', name='_work_edition_uc'),)
+class Review(core.Base):
+    #TODO: Update documentation:
+    """A rigorous book recommendation which has a winner and references
+    which candidates where involved in the decision"""
 
+    __tablename__ = "recommendations"
```

Comment:
Submit Rec... (submitter, topic, winner, contender[s],  review)

If we want a Topic *focus* (e.g. For high-schoolers) likely needs to be added in the Rec. table or there needs to be a rec_foci join table